lib/oli/delivery/sections.ex

```diff
@@ -5349,4 +5350,34 @@ defmodule Oli.Delivery.Sections do
     )
     |> Repo.all()
   end
+
+  def get_section_prompt_info(section_id) do
+    section_slug = Repo.get(Section, section_id).slug
+
+    instructors =
+      section_slug
+      |> fetch_instructors()
+      |> Enum.map(&%{name: &1.name, email: &1.email})
+
+    layout =
+      section_slug
+      |> fetch_ordered_container_labels()
+      |> Enum.map(&elem(&1, 1))
+
+    content =
+      section_slug
+      |> fetch_all_pages()
```

Comment:
Same thing here, we should add a new function to `SectionResourceDepot` that returns the needed info for all pages.

Reply:
Thanks for the feedback, fixed it in f91fe64cc8